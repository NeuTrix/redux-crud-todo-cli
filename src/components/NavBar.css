grid: {
		display: 'grid',
		gridTemplateAreas:`"logo welcome ... home todos login" `
	},

	logo: {
		gridArea: 'logo',
		fontSize: '0.5em',
	},

	welcome: {
		gridArea: 'welcome',
	},

	home: {
		gridArea: 'home',
	},

	todos: {
		gridArea: 'todos',
	},

	login: {
		gridArea: 'login',
	},

	logout:{
		gridArea: 'login',
		color: 'lime', 
	},

	base: {
		color: 'lightblue', 
		textAlign: 'center' 
	},

};