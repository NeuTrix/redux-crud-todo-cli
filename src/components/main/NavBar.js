import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

import AppBar from '@material-ui/core/AppBar';
import Divider from '@material-ui/core/Divider';
import Drawer from '@material-ui/core/Drawer';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import Toolbar from '@material-ui/core/Toolbar';
import { withStyles } from '@material-ui/core/styles';

import BrandLogo from './BrandLogo';
import NavigationLinks from './NavigationLinks';
import SearchBar from './SearchBar';

const propTypes = {
	classes: PropTypes.instanceOf(Object).isRequired,
	container: PropTypes.instanceOf(Object),
	isAuth: PropTypes.bool.isRequired,
	logout: PropTypes.func.isRequired,
	theme: PropTypes.instanceOf(Object).isRequired,
};

class NavBar extends Component {
	constructor(props) {
		super(props);
		this.state = {
			showMenu: false,
		};
		this.handleDrawerToggle = this.handleDrawerToggle.bind(this);
	}

	handleDrawerToggle(e) {
		e.preventDefault();
		this.setState(previousState => ({ showMenu: !previousState.showMenu }));
	}

	render() {
		const { showMenu } = this.state;
		const {
			isAuth,
			classes,
			container,
			logout,
			theme,
		} = this.props;

		return (
			<div className={`NavBar ${classes.root}`}>
				<AppBar className={classes.grid} position="fixed">
					<Toolbar>
						<IconButton
							color="inherit"
							aria-label="Open drawer"
							className={`menuButton ${classes.menuButton}`}
							onClick={this.handleDrawerToggle}
						>
							<MenuIcon />
						</IconButton>
					</Toolbar>
					<IconButton className={classes.brandLogo}>
						<NavLink to="/">
							<BrandLogo />
						</NavLink>
					</IconButton>
					<div className={classes.searchBar}>
						<SearchBar />
					</div>
				</AppBar>
				<nav className={classes.drawer}>
					<Drawer
						id="mainDrawer"
						container={container}
						variant="temporary"
						anchor={theme.direction === 'rtl' ? 'right' : 'left'}
						open={showMenu}
						classes={{ paper: classes.drawerPaper }}
						// Better open performance on mobile.
						ModalProps={{ keepMounted: true }}
						onClick={this.handleDrawerToggle}
					>
						<Divider />
						<NavigationLinks
							isAuth={isAuth}
							logout={logout}
							toggle={this.handleDrawerToggle}
						/>
						<Divider />
					</Drawer>
				</nav>
			</div>
		);
	}
}

const drawerWidth = 264;
const styles = theme => ({
	brandLogo: {
		display: 'none',
		gridArea: 'brandLogo',
		[theme.breakpoints.up('sm')]: {
			display: 'inherit',
		},
	},
	drawer: {
		[theme.breakpoints.up('sm')]: {
			flexShrink: 0,
			width: drawerWidth,
		},
	},
	drawerPaper: {
		width: drawerWidth,
	},
	grid: {
		display: 'grid',
		gridTemplateAreas: `
			" menuButton searchBar "
		`,
		gridTemplateColumns: '1fr 4fr',
		placeItems: 'center',
		[theme.breakpoints.up('sm')]: {
			gridTemplateAreas: `
			" brandLogo searchBar menuButton   "
		`,
			gridTemplateColumns: '2fr 4fr 1fr',
		},
	},
	menuButton: {
		color: theme.palette.contrast, // custom theme prop
		gridArea: 'menuButton',
		marginRight: 20,
	},
	root: { display: 'flex' },
	searchBar: {
		gridArea: 'searchBar',
		padding: theme.spacing.unit,
	},
});

NavBar.propTypes = propTypes;

export default withStyles(styles, { withTheme: true })(NavBar);
