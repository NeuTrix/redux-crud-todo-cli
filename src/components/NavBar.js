// vendor
import './NavBar.css';Â 
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
// +++++++++   +++++++++ 
import { connect } from 'react-redux';
// custom
import logo from '../assets/logo.svg';

import { fetchTodos } from '../actions/readActions';
import { logout } from '../actions/loginActions';

// +++++++++  CSS  +++++++++ 

const style = {

	navbox: {
		height: '3em',

	},

	welcome: {
		color:'steelblue',
		justifyContent: 'left',
	}
}



// +++++++++ COMPONENT  +++++++++ 

class NavBar extends Component {

	logout(e) {
		e.preventDefault();
		this.props.logout();
		this.context.router.history.push('/');
	}

	onClick(e){
		// e.preventDefault();
		this.props.fetchTodos();
	}

	render() {

		const { isAuthenticated, user }= this.props.authApi;

		const userLinks= (
			<div 
				id= 'logout_link' 
				className= 'ctr' 
				onClick= { this.logout.bind(this)} >
				<span >Log Out</span>
			</div>
		);

		const guestLinks= (
			<div>
				<div id= 'login_link' className= 'ctr'>
					<Link to= '/login' >Sign In</Link>
				</div>
				<div id= 'register_link' className= 'ctr'>
					<Link to= '/register' >Register</Link>
				</div>
			</div>
		);

		const welcome= (
			<div style= { style.welcome } > {`Welcome ${ user.username }!`} </div>
		);

		return (

			<nav className= 'engrBox paper' id= 'NavGrid' style= {style.navbox} >

				<div id= 'logo_link' className= 'ctr'>
					<Link to= '/' >
						<div className= "engr App-logo ctr fa fa-gg fa-2x" alt= "logo" />
					</Link>
				</div>

				<div id= 'welcome_link' className= 'ctr'> 
					{ isAuthenticated ? welcome : '' }  
				</div>

				<div id= 'home_link' className= 'ctr' >
					<Link to= '/' > Home </Link>
				</div>

				<div id= 'todos_link' className= 'ctr'>
					<Link to= '/todos' >
						<span onClick= { this.onClick.bind(this) }> Todos </span>
					</Link>
				</div>

					{ isAuthenticated ? userLinks : guestLinks }

			</nav>
		);
	}
}

// +++++++++ PROPS  +++++++++ 

NavBar.propTypes= {
	authApi: PropTypes.object.isRequired,
	logout: PropTypes.func.isRequired,
	placement: PropTypes.string,
	fetchTodos: PropTypes.func.isRequired
};

NavBar.contextTypes= {
	router: PropTypes.object.isRequired,
};

const mapStateToProps= (state) => {
	return { 
		authApi: state.authApi,
		user: state.authApi
	};
};

export default connect(mapStateToProps, { logout, fetchTodos })(NavBar);