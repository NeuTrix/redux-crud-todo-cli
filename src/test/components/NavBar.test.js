/* eslint-env jest, mocha, chai */
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from 'enzyme';
import NavBar from '../../components/main/NavBar';

describe('The NavBar Component w/ ReactDom', () => {
	const wrapper = render(
		<Router>
			<NavBar/>
		</Router>    
	);

	it('... renders the NavBar component', () => {
		// handle randomly generated class names from React
		expect(wrapper).toMatchSnapshot({
			class: expect.any(String)
		});
	});
});
