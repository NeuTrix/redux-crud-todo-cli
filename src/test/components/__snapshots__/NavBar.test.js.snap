// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] 1`] = `
<div
  class="nav-bar sc-EHOje hulMMD"
>
  <img
    alt="logo"
    class="logo sc-bZQynM jYoKCs"
    src="logo-white.png"
  >
    
  </img>
  <i
    class="fa"
  >
    
  </i>
  <div
    class="nav-section sc-bdVaJa gkdVAg"
  >
    <i
      class="fa"
    >
      
    </i>
    <ul
      class="sc-htpNat hwcvvy"
      id="menu"
    >
      <a
        auth="false"
        class="sc-bxivhb bhPHGk"
        href="/todos"
      >
        Todos
      </a>
      <a
        auth="false"
        class="sc-bxivhb bhPHGk"
        href="/"
      >
        Logout
      </a>
      <a
        auth="false"
        class="sc-ifAKCX ihoRgB"
        href="/register"
      >
        Register
      </a>
      <a
        auth="false"
        class="sc-ifAKCX ihoRgB"
        href="/login"
      >
        Login
      </a>
    </ul>
  </div>
</div>
`;
